crates/amaru/src/bin/amaru/main.rs

```diff
@@ -39,7 +40,7 @@ async fn main() -> miette::Result<()> {
     let args = Cli::parse();
 
     let result = match args.command {
-        Command::Daemon(args) => cmd::daemon::run(args, counter).await,
+        Command::Daemon(args) => cmd::daemon::run(args, counter, metrics.clone()).await,
```

Comment:
Note that this `counter` was merely me toying around passing a metric counter all-the-way down the ledger. What I had in mind for this was to become some sort of interface / handle to _metrics_ in general; possibly abstract behind traits and driven by the tracing setup.

Reply:
agreed, I had the same thought; I thought about trying to refactor it as part of this, but figured a lighter touch at least for the draft was better

Reply:
(I can take a pass at something like that in a follow up PR, if you want!)